### ISSUE TITLE
How to setup huggingface model and local hosted model?

### ISSUE THREAD

**Issue by alexsiu398** [lavague-ai/LaVague#307]
**Is your feature request related to a problem? Please describe.**
In previous release, it support huggingface and local hosted model. For recent release with world model and action engine, how to to config and setup huggingface and local hosted model?

**Describe the solution you'd like**
To use hugginface model and local hosted model.

**Comment by mbrunel** [comment]
Hello! For our current use cases we found that local models weren't quite good enough, which is why we don't encourage to use them anymore. However we're currently working on benchmarks and eval tools to help us understand how to make it work.
If you still want to use local llms and embeddings you can pass them directly to the classes.

for example:

```py
from llama_index.embeddings.huggingface import HuggingFaceEmbedding
from lavague.core.navigation import NavigationEngine
from lavague.drivers.selenium import SeleniumDriver

embedding = HuggingFaceEmbedding(
    model_name="BAAI/bge-small-en-v1.5"
)
driver = SeleniumDriver("https://news.ycombinator.com")
nav_engine = NavigationEngine(driver, embedding=embedding)
action = nav_engine.get_action("Enter hop inside the search bar and then press enter")
print(action)
```

This example allows you to have local embedding. And the same logic can be applied for the llm and multimodal llm.

Note though that local llms context window is often too small for our needs.
And the local multimodal llms are often just not good enough for the need of the world model.

**Comment by lyie28** [comment]
@alexsiu398  You can also see the customization guide: https://docs.lavague.ai/en/latest/docs/get-started/customization/

Let us know if this answers your question :)

**Comment by lyie28** [comment]
To give you a more concrete example, here's an example using a local LLM (zephyr-7b-alpha) and Local embedding model:

```python
from lavague.drivers.selenium import SeleniumDriver
from lavague.core import ActionEngine, WorldModel
from lavague.core.agents import WebAgent
from lavague.core.logger import AgentLogger
from llama_index.embeddings.huggingface import HuggingFaceEmbedding
from llama_index.llms.huggingface import HuggingFaceLLM

driver = SeleniumDriver(headless=True)

llm = HuggingFaceLLM(model_name="HuggingFaceH4/zephyr-7b-alpha")
embedding = HuggingFaceEmbedding(
    model_name="BAAI/bge-small-en-v1.5"
)

action_engine = ActionEngine(driver=driver, llm=llm, embedding=embedding)
world_model = WorldModel()

agent = WebAgent(world_model, action_engine)

agent.get("https://huggingface.co/")
agent.run("what is the top model", display=True)
```

I didn't change the multi-modal LLM to a local one as I am not aware of an open-source model with good enough performance at this point - so it still uses OpenAI credits just for the multi-modal LLM - of course, feel free to do your own tests with that!

Performance will vary for open-source models, sometimes they just don't perform well enough for our needs - let us know if you find any models that are working particularly well :)

**Comment by alexsiu398** [comment]
It seems like llama index only support ollama as an open source multimodal. But when I encounter some error after running the objective,like navigation error, is there any clues to figure out I should change llm,mm_llm or embedding? Or it's by trial and error?

**Comment by lyie28** [comment]
I have created a debugging guide to give some advice on how to identify the source of problems:
https://docs.lavague.ai/en/latest/docs/learn/debugging/

Please let me know if you find the guide helpful and if there is any information you'd like to see here.

**Comment by alexsiu398** [comment]
Is there any way to test multi-modal llm with huggingface model? The choice of multi-modal llm on ollama is limited. It would be great to have multi-modal llm with huggingface model.

**Comment by lyie28** [comment]
@alexsiu398 good point. We are limited by LlamaIndex's available multi-modal integrations. Let me investigate if there are any plans for a HuggingFace multi-modal module to be integrated at LlamaIndex. If not, we can open a `help-wanted` issue for now.
